subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -109,119 +111,192 @@ static int update_job_execution(struct mqtt_client *const client,
 	return ret;
 }
 
-
-static int aws_fota_on_publish_evt(struct mqtt_client *const client,
-				   const u8_t *topic,
-				   u32_t topic_len,
-				   u32_t payload_len)
+/**
+ * @brief Parsing an AWS IoT Job Execution response recived on $next/get MQTT
+ *	  topic or notify-next if it is an valid response the program state is
+ *	  updated.
+ *
+ * @param[in] client Connected MQTT client instance
+ * @param[in] payload_len Length of the payload going to be read out from the
+ *	      MQTT message.
+ *
+ * @return 1 If successful otherwise an negative error code is returned.
+ */
+static int get_job_execution(struct mqtt_client *const client,
+			     u32_t payload_len)
 {
-	int err;
+	int err = get_published_payload(client, payload_buf, payload_len);
 
-	LOG_INF("Received topic: %s", log_strdup(topic));
+	if (err) {
+		LOG_ERR("Error when getting the payload: %d", err);
+		return err;
+	}
+	/* Check if message received is a job. */
+	err = aws_fota_parse_DescribeJobExecution_rsp(payload_buf, payload_len,
+						      job_id, hostname,
+						      file_path,
+						     &execution_version_number);
+
+	if (err < 0) {
+		LOG_ERR("Error when parsing the json: %d", err);
+		return err;
+	} else  if (err == 1) {
+		LOG_DBG("Got only one field: %s", log_strdup(payload_buf));
+		LOG_INF("No queued jobs for this device");
+		return 1;
```

Comment:
is it ok to return success here even if there was no job found?

Reply:
Yes as we successfully checked if there was any job queued for the device. Even if it wasn't any jobs queued.